cmd/store.go

```diff
@@ -23,52 +24,46 @@ func (s *store) Get(IP string) (string, error) {
 	if role, ok := s.rolesByIP[IP]; ok {
 		return role, nil
 	}
+	log.Warnf("Using fallback role for IP %s", IP)
 	if s.defaultRole != "" {
-		log.Warnf("Using fallback role for IP %s", IP)
 		return s.defaultRole, nil
 	}
 	return "", fmt.Errorf("Unable to find role for IP %s", IP)
 }
 
 // OnAdd is called when a pod is added.
 func (s *store) OnAdd(obj interface{}) {
-	if pod, ok := obj.(*api.Pod); ok {
-		if pod.Status.PodIP != "" {
-			if role, ok := pod.Annotations[s.iamRoleKey]; ok {
-				s.mutex.Lock()
-				s.rolesByIP[pod.Status.PodIP] = role
-				s.mutex.Unlock()
-			}
+	pod := obj.(*api.Pod)
```

Comment:
It is not the case of an unknown message sent, store is just an implementation of a `cache.ResourceEventHandler`. The only reason why something unknown would be in the cache is if there is a problem in kubernetes internals, which is very similar to the issue I linked. Crashing is definitely not the way to handle cast errors IMO, we should log the issue and return.